src/backend/providers/playnite/PlayniteProvider.cpp

```diff
@@ -0,0 +1,306 @@
+        const bool should_update_progress = game_count % update_interval == 0;
+        if (!validate_game(game_info, components)) {
+            if (should_update_progress)
+                emit progressChanged(progress);
+            continue;
+        }
+
+        // By default, Playnite categorizes Windows games as "PC", which is handled by Pegasus default theme using the
+        // IBM logo. For a good out-of-the-box experience, this should be "Windows".
+        QString platform_name = components.platforms.at(game_info.platform_id);
+        if (platform_name == QLatin1String("PC"))
+            platform_name = QStringLiteral("Windows");
+        model::Collection& collection = *sctx.get_or_create_collection(platform_name);
+
+        PlayniteGameLaunchInfo launch_info = launch_helper.get_game_launch_info(game_info, components);
+
+        model::Game* game_ptr = nullptr;
+        if (launch_info.type == PlayniteGameLaunchInfo::LaunchType::STEAM) {
+            QString key = QStringLiteral("steam:%1").arg(game_info.game_id);
+            game_ptr = create_uri_game(key, collection, sctx);
+        } else {
+            const QFileInfo finfo(launch_info.path);
+            game_ptr = create_file_path_game(clean_abs_path(finfo), collection, sctx);
+        }
+
+        Q_ASSERT(game_ptr);
+
+        game_ptr->setLaunchCmd(launch_info.launch_cmd);
+        game_ptr->setLaunchWorkdir(launch_info.working_dir);
+        apply_game_fields(*game_ptr, components, game_info, playnite_dir);
+        sctx.game_add_to(*game_ptr, collection);
+        output.emplace_back(game_ptr);
+
+        if (should_update_progress)
+            emit progressChanged(progress);
+    }
+
+    return output;
+}
+
+bool PlayniteProvider::validate_game(const PlayniteGame& game, const PlayniteComponents& components) const
+{
+    if (game.name.isEmpty()) {
+        return false;
+    }
+
+    if (!game.installed) {
+        Log::info(display_name(), LOGMSG("Skipping not installed game %1").arg(game.name));
+        return false;
+    }
+
+    if (game.hidden) {
+        Log::info(display_name(), LOGMSG("Skipping hidden game %1").arg(game.name));
+        return false;
+    }
+
+    const auto platform_name_it = components.platforms.find(game.platform_id);
+    if (platform_name_it == components.platforms.end()) {
+        Log::info(display_name(), LOGMSG("No platform found for game %1, skipping...").arg(game.name));
+        return false;
+    }
+
+    if (game.play_action.type.isEmpty()) {
+        // Installed
+        if (game.play_action.path.isEmpty()) {
+            Log::info(display_name(), LOGMSG("No path found for game %1, skipping...").arg(game.name));
+            return false;
+        }
+
+        if (game.install_directory.isEmpty()) {
+            Log::info(display_name(), LOGMSG("No install directory found for game %1, skipping...").arg(game.name));
+            return false;
+        }
+    } else if (game.play_action.type == "1") {
```

Comment:
Hm, didn't you introduce an enum for this?

Reply:
I did, though it's only converted when we create a `PlayniteGameLaunchInfo` object.

I think I will move the enum out and see if there's a nice way to convert it.  I also should probably parse this as an int and not a string.

I like to keep my JSON models 1:1 with the data so that's why I didn't convert this into enum in the `PlayniteGame`.  It makes it easier to debug parsing issues imo.  But I agree it would make more sense to convert to an enum here first before doing these checks